include/nanobind/nb_eval.h

```diff
@@ -0,0 +1,97 @@
+/*
+    nanobind/nb_eval.h: Support for evaluating Python expressions and statements
+                        from strings and files
+
+    Adapted by Nico Schlömer from pybind11's eval.h with
+
+    Copyright (c) 2016 Klemens Morgenstern <klemens.morgenstern@ed-chemnitz.de> and
+                       Wenzel Jakob <wenzel.jakob@epfl.ch>
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#pragma once
+
+#include <nanobind/nanobind.h>
+
+#include <string>
+#include <utility>
+
+NAMESPACE_BEGIN(NB_NAMESPACE)
+
+/// \ingroup python_builtins
+/// Return a dictionary representing the global variables in the current execution frame,
+/// or ``__main__.__dict__`` if there is no frame (usually when the interpreter is embedded).
+inline dict globals() {
+    PyObject *p = PyEval_GetGlobals();
+    return borrow<dict>(p ? p : module_::import_("__main__").attr("__dict__").ptr());
+}
+
+enum eval_mode {
+    /// Evaluate a string containing an isolated expression
+    eval_expr,
```

Comment:
let's set this equal to `Py_eval_input` (ditto for the other cases below). That way, the `switch` statement in the `eval()` implementation can disappear.